lib/workers/repository/process/lookup/abandonment.ts

```diff
@@ -0,0 +1,32 @@
+import { DateTime } from 'luxon';
+import type { ReleaseResult } from '../../../../modules/datasource/types';
+import { toMs } from '../../../../util/pretty-time';
+import type { LookupUpdateConfig } from './types';
+
+export function calculateAbandonment(
+  releaseResult: ReleaseResult,
+  config: LookupUpdateConfig,
+): ReleaseResult {
+  const { abandonmentThreshold } = config;
+  if (!abandonmentThreshold) {
+    return releaseResult;
+  }
+
+  const abandonmentThresholdMs = toMs(abandonmentThreshold);
+  if (!abandonmentThresholdMs) {
+    return releaseResult;
+  }
+
+  const { bumpedAt } = releaseResult;
+  if (!bumpedAt) {
+    return releaseResult;
+  }
+  const bumpedAtDate = DateTime.fromISO(bumpedAt);
+
+  const abandonmentDate = bumpedAtDate.plus({
+    milliseconds: abandonmentThresholdMs,
+  });
+  const isAbandoned = abandonmentDate < DateTime.now();
```

Comment:
should we use local time? like we do in other places?

Reply:
I don't think datasource packages are I. Local time zone so that doesn't seem to make sense. It's all relative (also unlikely to be impacted by hours)